### PR TITLE
jdbc solution

### PR DESCRIPTION
**Pull request by Aalderaan**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String query = "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = false; ";
+        Manufacturer manufacturer = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturer = connection.prepareStatement(query)) {
+            getManufacturer.setLong(1, id);
+            ResultSet resultSet = getManufacturer
+                    .executeQuery();
```
Comment (LayRipper) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:96]:
```suggestion
            ResultSet resultSet = getManufacturer.executeQuery();
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement = connection
+                        .prepareStatement(deleteRequest, Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setLong(1, id);
+            return createManufacturerStatement.executeUpdate() >= 1;
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Can't delete manufacturer from DB", throwables);
```
Comment (LayRipper) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:115]: specify id of the manufacturer in error message pls

**Review by LayRipper — CHANGES_REQUESTED** [review]

**Review by LayRipper — APPROVED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        String query = "SELECT * FROM manufacturers WHERE is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllManufacturers = connection.createStatement()) {
+            ResultSet resultSet = getAllManufacturers.executeQuery(query);
+            while (resultSet.next()) {
+                Manufacturer manufacturer = setManufacturerFromResultSet(resultSet);
+                manufacturers.add(manufacturer);
+            }
+        } catch (SQLException throwables) {
```
Comment (RomanDubovskyi) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:29]:
```suggestion
        } catch (SQLException e) {
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            throw new DataProcessingException("Can't get all manufacturers from DB", throwables);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String insertFormatRequest =
```
Comment (RomanDubovskyi) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:37]:
```suggestion
        String updateRequest =
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't insert manufacturer with id: "
+                    + manufacturer.getId() + " to DB", throwables);
```
Comment (RomanDubovskyi) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:73]:
```suggestion
                    + manufacturer + " to DB", throwables);
```
If you didn't insert it, manufacturer has no id

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        }
+        return Optional.ofNullable(manufacturer);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement = connection
+                        .prepareStatement(deleteRequest, Statement.RETURN_GENERATED_KEYS)) {
```
Comment (RomanDubovskyi) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:101]: do we need RETURN_GENERATED_KEYS

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**